### PR TITLE
Added Proxy Authentication option in Client options constructor

### PR DESCRIPTION
**Pull request by danielR2001**
## Description

Added Puppeteer proxy authentication option in Client options constructor.
```
async initialize() {
   //   code...
   if (this.options.proxyAuthentication !== undefined) {
      await page.authenticate(this.options.proxyAuthentication);
   }
   //   code...
}
```
      
## Motivation and Context

I wanted to set up a proxy server in puppeteer but couldn't authenticate with it.

## Types of changes

- [ ] Dependency change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly (index.d.ts).

### CODE REVIEW

**Review by tuyuribr — CHANGES_REQUESTED** [review]
Sounds like a cool feat but you can't change all files and submit in a PR, please remove any unrelated changes

**Comment by danielR2001** [comment]
All the changed files are the documentation auto generate 😅

**Comment by danielR2001** [comment]
I removed the documentation generate from the merge

**Comment by danielR2001** [comment]
why no one reviewing this pr?

**Comment by tuyuribr** [comment]
Sorry for the delay, could you put an example in the example file and add the param to the index.d.ts ? Then I think we can merge

**Review by tuyuribr — CHANGES_REQUESTED** [review]
Index.d.ts

File: src/util/Injected.js [hunk 1]
```diff
@@ -17,15 +17,15 @@ exports.ExposeStore = (moduleRaidStr) => {
...
+    window.Store.MediaPrep = window.mR.findModule('prepRawMedia')[0];
     window.Store.MediaObject = window.mR.findModule('getOrCreateMediaObject')[0];
     window.Store.NumberInfo = window.mR.findModule('formattedPhoneNumber')[0];
     window.Store.MediaTypes = window.mR.findModule('msgToMediaType')[0];
     window.Store.MediaUpload = window.mR.findModule('uploadMedia')[0];
     window.Store.MsgKey = window.mR.findModule((module) => module.default && module.default.fromString)[0].default;
     window.Store.MessageInfo = window.mR.findModule('sendQueryMsgInfo')[0];
     window.Store.OpaqueData = window.mR.findModule(module => module.default && module.default.createFromData)[0].default;
-    window.Store.QueryExist = window.mR.findModule('queryExists')[0].queryExists;
+    window.Store.QueryExist = window.mR.findModule('queryExists')[0] ? window.mR.findModule('queryExists')[0].queryExists : window.mR.findModule('queryExist')[0].queryWidExists;
```
Comment (PurpShell) [inline src/util/Injected.js:28]: why do these exist? seperate into different branches and rebase

**Review by purpshell — CHANGES_REQUESTED** [review]
requesting changes

File: src/util/Injected.js [hunk 2]
```diff
@@ -17,15 +17,15 @@ exports.ExposeStore = (moduleRaidStr) => {
     window.Store.Invite = window.mR.findModule('resetGroupInviteCode')[0];
     window.Store.InviteInfo = window.mR.findModule('queryGroupInvite')[0];
     window.Store.Label = window.mR.findModule('LabelCollection')[0].LabelCollection;
-    window.Store.MediaPrep = window.mR.findModule('MediaPrep')[0];
+    window.Store.MediaPrep = window.mR.findModule('prepRawMedia')[0];
```
Comment (PurpShell) [inline src/util/Injected.js:20]:
```suggestion
    window.Store.MediaPrep = window.mR.findModule('MediaPrep')[0];
```

**Comment by PurpShell** [comment]
Sorry, @danielR2001. I'll close this PR for your inactivity. #2143 has opened, credits will go to you

**Pull request closed without merging**